node/sovereignNode_test.go

```diff
@@ -0,0 +1,354 @@
+package node_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/api"
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/node"
+	"github.com/multiversx/mx-chain-go/node/mock"
+	"github.com/multiversx/mx-chain-go/state"
+	"github.com/multiversx/mx-chain-go/testscommon"
+	stateMock "github.com/multiversx/mx-chain-go/testscommon/state"
+	trieMock "github.com/multiversx/mx-chain-go/testscommon/trie"
+	"github.com/multiversx/mx-chain-go/vm/systemSmartContracts"
+	vmcommon "github.com/multiversx/mx-chain-vm-common-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewSovereignNode(t *testing.T) {
+	n, err := node.NewNode()
+	require.Nil(t, err)
+	require.NotNil(t, n)
+
+	sn, err := node.NewSovereignNode(n)
+	require.Nil(t, err)
+	require.NotNil(t, sn)
+}
```

Comment:
1. Add t.Paralell.
2. You can remove `TestNewSovereignNode_NilNodeShouldError` and test that in this func as well. We usually use a format like this:

```go
	t.Run("nil server should error", func(t *testing.T) {
		t.Parallel()

		hs, err := NewHttpServer(nil)
		require.Equal(t, apiErrors.ErrNilHttpServer, err)
		require.Nil(t, hs)
	})
```
3. When checking with nil node input, check that the instance is nil and the error is ErrNilNode